### PR TITLE
Adherence component

### PR DESCRIPTION
**Pull request by collijk**
Warning: not tested

### CODE REVIEW

File: src/vivarium_csu_zenon/components/treatment/parameters.py [hunk 1]
```diff
@@ -29,6 +29,13 @@
 LIFESTYLE = 'lifestyle_intervention'
 
 
+SINGLE_NO_CVE = (0, 0)
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/parameters.py]: what does CVE stand for?

File: src/vivarium_csu_zenon/components/treatment/treatment_coverage.py [hunk 2]
```diff
@@ -145,4 +143,69 @@ def load_treatment_type_p(builder):
...
+
+        pop_status = self.population_view.subview(self.columns_required).get(propensity.index)
+        ihd = pop_status[project_globals.IHD_MODEL_NAME] != project_globals.IHD_SUSCEPTIBLE_STATE_NAME
+        stroke = (pop_status[project_globals.ISCHEMIC_STROKE_MODEL_NAME]
+                  != project_globals.ISCHEMIC_STROKE_SUSCEPTIBLE_STATE_NAME)
+        on_statin = (pop_status[parameters.STATIN_HIGH] != 'none') | (pop_status[parameters.STATIN_LOW] != 'none')
+        num_drugs = sum([on_statin, pop_status[parameters.FIBRATES], pop_status[parameters.EZETIMIBE]])
+        on_fdc = pop_status[parameters.FDC]
+
+        had_cve = ihd | stroke
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment_coverage.py]: cve must mean "cardiovascular event"!

File: src/vivarium_csu_zenon/components/treatment/treatment_coverage.py [hunk 3]
```diff
@@ -145,4 +143,69 @@ def load_treatment_type_p(builder):
...
+        single_med = (num_drugs == 1) | on_fdc
+        multi_med = num_drugs > 1
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment_coverage.py]: Did you intend single_med and multi_med to be mutually exclusive?  Instead could say `multiple_pills = (num_drugs > 1) & ~on_fdc`.

File: src/vivarium_csu_zenon/components/treatment/treatment_coverage.py [hunk 4]
```diff
@@ -145,4 +143,69 @@ def load_treatment_type_p(builder):
...
+
+        p_adherent.loc[~had_cve & single_med] = self.p_adherent[parameters.SINGLE_NO_CVE]
+        p_adherent.loc[~had_cve & multi_med] = self.p_adherent[parameters.MULTI_NO_CVE]
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment_coverage.py]: as written, I think you will incorrectly overwrite some rows where individuals are taking ezetimibe and statins together with a fixed dose combination pill.

File: src/vivarium_csu_zenon/components/treatment/treatment_coverage.py [hunk 5]
```diff
@@ -145,4 +143,69 @@ def load_treatment_type_p(builder):
...
+        p_adherent.loc[had_cve & single_med] = self.p_adherent[parameters.SINGLE_CVE]
+        p_adherent.loc[had_cve & multi_med] = self.p_adherent[parameters.MULTI_CVE]
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment_coverage.py]: similar issue to line 197.  could switch to `had_cve & ~single_med` to fix

File: src/vivarium_csu_zenon/components/treatment/treatment_coverage.py (hunk 3 above)
Comment (collijk) [reply src/vivarium_csu_zenon/components/treatment/treatment_coverage.py]: Yes, good catch.

File: src/vivarium_csu_zenon/components/treatment/parameters.py (hunk 1 above)
Comment (rmudambi) [reply src/vivarium_csu_zenon/components/treatment/parameters.py]: Cardiovascular event

**Pull request merged**